process/transactionProcessor.go

```diff
@@ -421,6 +422,16 @@ func (tp *TransactionProcessor) computeTransactionStatus(tx *transaction.ApiTran
 		return data.TxStatusUnknown
 	}
 
+	// sanity checks
+	senderAddress, err := tp.pubKeyConverter.Decode(tx.Sender)
+	if err != nil {
+		return transaction.TxStatusFail
```

Comment:
If sender is _**metachain**_ (e.g. reward transactions), works as expected?

https://testnet-gateway.multiversx.com/transaction/191a0bf2d29559181da9d93bfd35f8b9f136122006d10b51096fec6be65fb25d

On failure to decode, perhaps `TxStatusUnknown`?

Reply:
It seems that for reward transactions, `process-status` currently works fine:

https://testnet-gateway.multiversx.com/transaction/191a0bf2d29559181da9d93bfd35f8b9f136122006d10b51096fec6be65fb25d/process-status

Reply:
whoa, thanks for the input. Will add these examples and will come up with a fix

Reply:
done